qiskit/mapper/_layout.py

```diff
@@ -69,6 +75,19 @@ def __delitem__(self, key):
         dict.__delitem__(self, self[key])
         dict.__delitem__(self, key)
 
+    @staticmethod
+    def checktype(thing):
+        """Checks if thing is a valid type"""
+        if thing is None:
+            return
```

Comment:
why is None a valid input to `__setitem__`?

Reply:
Because it is convenient when Layouts is defined as `[(q0, 0), (q1, 0), None, (q1, 1), (q0, 1)]`.